azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -707,3 +720,43 @@ func resourceKubernetesClusterNodePoolDelete(d *schema.ResourceData, meta interf
 
 	return nil
 }
+
+func upgradeSettingsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"max_surge": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+}
+
+func expandUpgradeSettings(input []interface{}) *containerservice.AgentPoolUpgradeSettings {
+	if len(input) == 0 {
+		return nil
```

Comment:
this should return an empty block - else it's not possible to reset these (e.g. go from having them specified to not)

```suggestion
		return &containerservice.AgentPoolUpgradeSettings{}
```